app/controllers/transcriptions_controller.rb

```diff
@@ -13,21 +19,24 @@ def index
   def show
     @transcription = Transcription.find(params[:id])
     authorize @transcription
+
+    if TranscriptionPolicy.new(current_user, @transcription).has_update_rights?
+      @transcription.lock(current_user)
+    end
+
+    response.set_header('Last-Modified', @transcription.updated_at.rfc2822)
     render jsonapi: @transcription
   end
 
   def update
     @transcription = Transcription.find(params[:id])
-    raise ActionController::BadRequest if type_invalid?
-    raise ActionController::BadRequest unless whitelisted_attributes?
-
-    if approving?
-      authorize @transcription, :approve?
-    else
-      authorize @transcription
-    end
+    validate_update
+    authorize_update
 
     update_attrs['updated_by'] = current_user.login
+    update_attrs['locked_by'] = current_user.login
+    update_attrs['lock_timeout'] = DateTime.now + 3.hours
```

Comment:
not blocking: I'd probably make this configurable via ENV var with your sensible default
```suggestion
    update_attrs['lock_timeout'] = DateTime.now + ENV.fetch('MODEL_LOCK_TIME', 3.hours)
```

Reply:
Since we're using rails credentials in tove, would it make sense to store this value in the credentials file? From my understanding, the credentials file is meant to serve the same purpose as environment variables do.

Reply:
the issue with that is it requires a PR to deploy the change etc. So code review etc. If it's just the deployed ENV var with a sensible default we can modify the deploy env to enable the different value.